pumpkin-lib/src/propagators/arithmetic/maximum.rs

```diff
@@ -67,23 +69,120 @@ impl<ElementVar: IntegerVariable, Rhs: IntegerVariable> Propagator
         &self,
         mut context: PropagationContextMut,
     ) -> PropagationStatusCP {
+        // This is the constraint that is being propagated:
+        // max(a0, a1, ..., an-1) = rhs
+
+        // Propagate bounds on the variables based on the rhs:
+        // Rule 1.
+        // UB(a_i) <= UB(rhs)
+
+        // Propagate bounds on the rhs variable:
+        // Rule 2.
+        // LB(rhs) >= max{LB(a_i)}.
+        // Rule 3.
+        // max_ub = max{UB(a_i)}.
+        // if all UB(a_i) <= max_ub, then UB(rhs) <= max_ub.
```

Comment:
Does this make sense? Given that `max_ub=max{UB(a_i)}`, the predicate `UB(a_i) <= max_ub` is trivially true.

Reply:
Fixed!